vllm/compilation/backends.py

```diff
@@ -624,9 +624,12 @@ def __call__(self, graph: fx.GraphModule, example_inputs) -> Callable:
         ]
 
         # index of tensors that have symbolic shapes (batch size)
+        # for weights and static buffers, they will have concrete shapes,
+        # and therefore x.numel() will be an integer.
         self.sym_tensor_indices = [
             i for i, x in enumerate(fake_args)
             if isinstance(x, torch._subclasses.fake_tensor.FakeTensor)
+            and not isinstance(x.numel(), int)
         ]
```

Comment:
Looks like there is a more official way to check (although experimental) https://github.com/pytorch/pytorch/blob/9e0437a04a2fbb546552f0382fe75941f5bc4fb5/torch/fx/experimental/symbolic_shapes.py#L1867-L1875

```
from torch.fx.experimental.symbolic_shapes import is_symbolic

self.sym_tensor_indices = [
    i for i, x in enumerate(fake_args)
    if isinstance(x, torch._subclasses.fake_tensor.FakeTensor)
    and any(is_symbolic(d) for d in x.size())
]
```

Reply:
wow didn't know this. thanks!
fixed in [d5c42f2](https://github.com/vllm-project/vllm/pull/12191/commits/d5c42f2b20a6454f55e8fcb0da8942d5deccbb88)